apps/site/src/app/admin/layout/AdminLayout.tsx

```diff
@@ -30,19 +33,29 @@ function AdminLayout({ children }: PropsWithChildren) {
 	const authorized = isAdminRole(role);
 
 	if (!loggedIn) {
-		redirect("/login");
+		router.push("/login");
 	} else if (!authorized) {
-		redirect("/unauthorized");
+		router.push("/unauthorized");
 	}
 
 	return (
-		<UserContext.Provider value={identity}>
-			<AppLayout
-				content={children}
-				navigation={<AdminSidebar />}
-				breadcrumbs={<Breadcrumbs />}
-			/>
-		</UserContext.Provider>
+		<SWRConfig
+			value={{
+				onError: (err) => {
+					if (axios.isAxiosError(err)) {
+						router.push("/login");
+					}
+				},
```

Comment:
Issue: we don't necessarily want to redirect to Login on every API request error, e.g. if the server has a hiccup, we should let SWR retry. The Login redirect would make sense for only status 401.
Thought: sorry for the mixup, but I think specifying a function for `shouldRetryOnError` would be a little more appropriate to prevent retries even though the redirect should theoretically happen immediately.